docs/README.md

```diff
@@ -0,0 +1,142 @@
+anything. Another word might have been better, but sync is how we all talked
+about it. Language is weird.
+
+[sync_task]: /lib/tasks/redshift.rake
+[sync_model]: /app/models/sync.rb
+[sync_list]: /app/views/syncs/index.html.haml
+
+### Sync States
+
+The `Sync` model has a `state` column which is managed by the
+[`SyncMicroMachine` class][sync_mm]. The `Sync` progresses along this graph:
+
+![Sync graph][sync_graph]
+
+[sync_mm]: /app/models/sync_micro_machine.rb
+[sync_graph]: /docs/SyncMicroMachine_graph.png
+
+### Sync Workers
+
+Exporting data to Redshift is accomplished by workers. Every 10 minutes when the
+sync task is run it enqueues a [`SyncManagementJob`][sync_mgmt] which does one
+of these things:
+
+* creates a new `Sync`
+* kicks off copying when a `Sync` is ready for it
+* nothing
+
+When there are no in-progress `Sync` records, then [a new one is
+started][sync_start] and that enqueues a [`StartSyncJob`][sync_start_job]. This
+job will either skip if there's nothing to do or split up the data into
+manageable chucks and enqueue [`OrganizationExportJobs`][org_export_job] for
+each part.
+
+The jobs to export a batch of `Organization` records are processed
+asynchronously and that happens on it's own queue because it's very memory
+intensive. This is where we resolve and flatten our relational data using the
+aptly named [`OrganizationResolver`][org_resolver].
+
+The second thing the management job could do is to [call `copy`][sync_copy] on
+an existing `Sync` record. Copying a Sync is another way of saying that you're
+going to finish it, thus this method enqueues a
+[`FinishSyncJob`][sync_finish_job]. This job collaborates with the
+[`DataWarehouse`][data_warehouse] to truncate and copy all data from S3 to
+Redshift.
+
+[sync_mgmt]: /app/jobs/sync_management_job.rb
+[sync_start]: /app/models/sync.rb#L8
+[sync_start_job]: /app/jobs/start_sync_job.rb
+[org_export_job]: /app/jobs/organization_export_job.rb
+[org_resolver]: /app/models/organization_resolver.rb
+[sync_copy]: /app/models/sync.rb#L25
+[sync_finish_job]: /app/jobs/finish_sync_job.rb
+[data_warehouse]: /app/models/data_warehouse.rb
+
+## Resolving Organizations
+
+Yes Bearden imports and exports, but the thing that makes it special is the way
+that it models data and resolves `Organization` records. Rather than simply
+having an email column on the organizations table, for example, we have a table
+of email addresses that point to their organization. This means the actual
+`Organization` has very little on it and serves mostly as a way to relate all
+the details of an entity. Each of these details points back to the `Source` that
+created it, which means that we always know which details are most accurate.
+
+Evaluating this accuracy happens at export time, during the process of syncing
+with Redshift. Waiting until this point in the process means means that we
+delay the flattening to as late as possible and reflect the most up-to-date
+information as possible.
```

Comment:
rather than "up-to-date", "information from the highest-ranked source." This, or below in Rankables, might be a good place to introduce the concept of each Source having a rank for each field, which is alluded to elsewhere but not explicitly stated. The per-field aspect also makes Bearden special.